### PR TITLE
util-lib: fix return value in mkdir_parents()

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/basic/mkdir.c [hunk 1]
```diff
@@ -116,8 +115,8 @@ int mkdir_parents_internal(const char *prefix, const char *path, mode_t mode, mk
                         continue;
 
                 r = _mkdir(t, mode);
-                if (r < 0 && errno != EEXIST)
-                        return -errno;
```
Comment (poettering) [inline src/basic/mkdir.c:120]: hmm, i think the old code was right, _mkdir is after a function parameter, where libc mkdir might be passed which returns errors in errno

Comment (keszybz) [reply src/basic/mkdir.c:120]: See #7659. I think this might be related.

**Comment by keszybz** [comment]
Updated, PTAL. I don't like the name of the new function, but I didn't have a better idea.

**Comment by keszybz** [comment]
`artful-i386`: `The file /dev/loop11p1 does not exist and no size was specified.`
`xenial-amd64`: `KVM: entry failed, hardware error 0x0`

**Comment by yuwata** [comment]
Does this fix #7659?

**Pull request merged**